OpenMTRDemo/Filters/IsolateDialsFilter.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using OpenCvSharp;
+using OpenMTRDemo.Forms;
+using OpenMTRDemo.Models;
+using System.Windows.Forms;
+using System.Threading;
+
+namespace OpenMTRDemo.Filters
+{
+    public partial class IsolateDialsFilter : BaseFilter
+    {
+        private List<CircleSegment> _circleList;
+
+        public IsolateDialsFilter(ExpandedImageForm Editor, MeterImage Meter, List<CircleSegment> circleList = null)
+        {
+            this.Editor = Editor;
+            this.Meter = Meter;
+            _circleList = (circleList == null) ? new List<CircleSegment>() : circleList;
+            FilterName = "Isolate Dials";
+            Type = 3;
+            InitializeComponent();
+        }
+
+        public override void ApplyFilter(Mat image)
+        {
+            if (image.Type() == MatType.CV_8UC1)
+            {
+                _circleList = new List<CircleSegment>();
+                CircleSegment[] circles = Cv2.HoughCircles(image, HoughMethods.Gradient, 1, (Math.Min(image.Height, image.Width) / 12), 250, 100, (Math.Min(image.Height, image.Width) / 20), (Math.Min(image.Height, image.Width) / 3));
+                foreach (CircleSegment circle in circles)
```

Comment:
I feel like this logic can be simplified quite a bit by going ahead and checking to make sure the center XY coordinates are not the same throughout the list and then proceed to create the boundaries.